nmi_mysql/nmi_mysql.py

```diff
@@ -9,9 +9,42 @@
 from queue import Queue
 
 
+CONFIG_KEYS = ['host', 'user', 'password', 'db', 'port']
+MAX_POOL_SIZE = 10
+
+
 class DB(object):
 
-    def __init__(self, conf, autoconnect=False):
+    def __init__(self, conf):
+        for c in CONFIG_KEYS:
+            if c not in conf:
+                raise ValueError('Invalid config object')
+
+        self.conf = conf
+        self.max_pool_size = conf.get('max_pool_size', MAX_POOL_SIZE)
+        self._initialize_pool()
+
+    def _initialize_pool(self):
+        self.pool = Queue(maxsize=self.max_pool_size)
+
+        for _ in range(0, self.max_pool_size):
+            self.pool.put(_Connection(self.conf))
+
+    def query(self, query, params=None):
+        con = self.pool.get()
+        con.connect()
```

Comment:
This should only be done when con is not yet connected (which by default should already be connected if you get it from their pool?

For example: http://stackoverflow.com/questions/22699807/python-mysql-using-pymysql-auto-reconnect

Another thought, Instead of rewriting this, why don't we just use a proven library that has pooling support?

ex: 
http://www.sqlalchemy.org/
https://launchpad.net/pysqlpool
http://stackoverflow.com/questions/98687/what-is-the-best-solution-for-database-connection-pooling-in-python